lib/driver.js

```diff
@@ -435,6 +442,43 @@ class AndroidDriver extends BaseDriver {
     }
   }
 
+  async setSharedPreferences () {
+    let sharedPrefs = this.opts.sharedPreferences;
```

Comment:
does the builder mutate `sharedPrefs`? if so it will also mutate `this.opts.sharedPreferences`. in that case we'd want to clone rather than assign. but i don't know how the builder works.